pkg/resources/stage.go

```diff
@@ -357,57 +297,81 @@ func UpdateStage(d *schema.ResourceData, meta interface{}) error {
 		encryption := d.Get("encryption")
 		q := builder.ChangeEncryption(encryption.(string))
 		if err := snowflake.Exec(db, q); err != nil {
-			return fmt.Errorf("error updating stage encryption on %v", d.Id())
+			return diag.Errorf("error updating stage encryption on %v", d.Id())
 		}
 	}
+
 	if d.HasChange("file_format") {
 		fileFormat := d.Get("file_format")
 		q := builder.ChangeFileFormat(fileFormat.(string))
 		if err := snowflake.Exec(db, q); err != nil {
-			return fmt.Errorf("error updating stage file formaat on %v", d.Id())
+			return diag.Errorf("error updating stage file format on %v", d.Id())
 		}
 	}
+
 	if d.HasChange("copy_options") {
 		copyOptions := d.Get("copy_options")
 		q := builder.ChangeCopyOptions(copyOptions.(string))
 		if err := snowflake.Exec(db, q); err != nil {
-			return fmt.Errorf("error updating stage copy options on %v", d.Id())
+			return diag.Errorf("error updating stage copy options on %v", d.Id())
 		}
 	}
+
 	if d.HasChange("comment") {
 		comment := d.Get("comment")
 		q := builder.ChangeComment(comment.(string))
 		if err := snowflake.Exec(db, q); err != nil {
-			return fmt.Errorf("error updating stage comment on %v", d.Id())
+			return diag.Errorf("error updating stage comment on %v", d.Id())
 		}
 	}
 
-	tagChangeErr := handleTagChanges(db, d, builder)
-	if tagChangeErr != nil {
-		return tagChangeErr
+	if d.HasChange("tag") {
```

Comment:
is there a reason for moving the tag diff logic here?

Reply:
So, tags were one of the things I could change in the Update method, so I did it. `handleTagChanges` is the old implementation using the snowflake package.